src/Animals/Animal.ts

```diff
@@ -0,0 +1,21 @@
+import { AnimalNames } from '../Enums/AnimalNamesEnum';
+
+export abstract class Animal {
+  constructor(
+    protected name: AnimalNames,
+    protected value?: number,
```

Comment:
Is it really optional?

Reply:
yes, value is optional, applies only to herd animals, like cow and rabbit. Predators like wolf and fox will not use this field.